### PR TITLE
feat: add optional tqdm progress bars to TimesFMFinetuner

### LINKED ISSUE #289: Add tqdm progress bars to TimesFMFinetuner (finetuning_torch.py)
### Problem
`TimesFMFinetuner` currently trains silently. Long runs give no per-batch feedback.

### Proposal
Add an optional `tqdm` progress bar around the train/val DataLoader loops (rank 0 only).  
Introduce a single flag in `FinetuningConfig`:
```python
progress_bar: bool = True
```

Default remains True (can be set to False to keep old behavior).

### Implementation details

- Wrap loops in _train_epoch and _validate with tqdm when config.progress_bar and rank == 0.
- Show running average loss via set_postfix every log_every_n_steps.

### Backward compatibility

Setting progress_bar=False yields the exact current behavior (no bars). Otherwise, there’s only additional stdout output.

### Testing
- Tiny smoke test to ensure training runs with progress_bar=True without crashing.

Happy to submit a PR with the change. Let me know if you prefer a different flag name or default!

### PR DESCRIPTION
**Pull request by PouyaGhahramanian**
### What
Adds an optional `tqdm` progress bar around the train & validation loops in `TimesFMFinetuner`.

### Why
Long fine-tuning jobs provide no per-batch feedback right now. Progress bars improve UX without changing training logic.

### How
- `FinetuningConfig` now includes `progress_bar: bool = True`.
- Wrap the DataLoader iterators in `_train_epoch`/`_validate` with `tqdm` when `config.progress_bar` and `rank == 0`.
- Display running average loss every `log_every_n_steps` via `set_postfix`.

### Backward compatibility
Setting `progress_bar=False` disables the bars; otherwise only stdout changes.

Closes #289.

### CODE REVIEW

**Comment by MonkeyCode-AI** [comment]
⏳ MonkeyCode-AI 正在分析，请稍等片刻...

**Comment by MonkeyCode-AI** [comment]
⏳ MonkeyCode-AI 正在分析，请稍等片刻...
